packages/typeahead/src/react/index.tsx

```diff
@@ -135,16 +160,19 @@ const Typeahead: TypeaheadFieldComponent = props => {
             ? { label: searchTerm as string }
             : selectedItem || undefined
         )
-      setInputItems(
-        options.filter(({ label }) =>
-          `${label}`
-            .toLowerCase()
-            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-            .startsWith((inputValue as string).toLowerCase())
-        )
-      )
+      setInputItems(filterFunction(options, inputValue))
     }
   })
+  React.useEffect(() => {
+    onChange &&
+      onChange(
+        null,
+        activeItem as {
```

Comment:
Does downshift have a generically-typed `useCombobox` by chance?  That would probably enable the avoidance of the cast here.

If not, or perhaps in addition, I might create an interface type like `DropdownItemProps` (or whatever might seem a more appropriate name) so that this shape doesn't need redefined multiple times and that the relationship in code will be more apparent.

Reply:
selectedItem renamed activeItem can actually be what you want. As for a reusable interface sure I could do that will still require cast.